File: backend/app/schemas/live.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class CreateRoomRequest(BaseModel):
    subject: str
    topic: Optional[str] = None
    student_id: Optional[str] = None


class RoomResponse(BaseModel):
    room_id: str
    provider: str
    provider_room_id: Optional[str] = None
    token: Optional[str] = None
    status: str
    created_at: datetime


class EndSessionRequest(BaseModel):
    recording_url: Optional[str] = None


class SessionDetailResponse(BaseModel):
    id: str
    room_id: str
    subject: str
    topic: Optional[str] = None
    status: str
    provider: str
    recording_url: Optional[str] = None
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None
    created_at: datetime

File: frontend/src/services/live.ts
import { api } from "./api"

export interface LiveRoom {
  room_id: string
  provider: string
  provider_room_id: string | null
  token: string | null
  status: string
  created_at: string
}

export interface LiveSession {
  id: string
  room_id: string
  subject: string
  topic: string | null
  status: string
  provider: string
  recording_url: string | null
  started_at: string | null
  ended_at: string | null
  created_at: string
}

export const liveService = {
  createRoom: (data: { subject: string; topic?: string; student_id?: string }) =>
    api.post<LiveRoom>("/live/rooms", data),

  getRoom: (roomId: string) =>
    api.get<LiveSession>(`/live/rooms/${roomId}`),

  startSession: (roomId: string) =>
    api.post<LiveSession>(`/live/rooms/${roomId}/start`),

  endSession: (roomId: string, recordingUrl?: string) =>
    api.post<LiveSession>(`/live/rooms/${roomId}/end`, { recording_url: recordingUrl }),

  listSessions: (role = "tutor", skip = 0, limit = 50) => {
    const params = new URLSearchParams({ role, skip: String(skip), limit: String(limit) })
    return api.get<LiveSession[]>(`/live?${params}`)
  },
}
